### PR TITLE
feat: add editor and interactive viewer to default role options

### PR DESCRIPTION
**Pull request by ZeRego**
Closes: #5507 

### Description:

Note: not adding developer and admin as that options since they have high level of permissions and access. Setting these roles as default role by mistake could cause security issues for that org. We can reconsider this in the future.

Captura de ecrã 2023-06-22, às 11 49 22

### CODE REVIEW

File: packages/frontend/src/components/UserSettings/AllowedDomainsPanel/index.tsx [hunk 1]
```diff
@@ -14,6 +14,16 @@ import { useProjects } from '../../../hooks/useProjects';
 import { isValidEmailDomain } from '../../../utils/fieldValidators';
 
 const roleOptions = [
+    {
```
Comment (owlas) [inline packages/frontend/src/components/UserSettings/AllowedDomainsPanel/index.tsx:17]: so the api accepts any role as the default role? Do we want to change that?

Comment (ZeRego) [reply packages/frontend/src/components/UserSettings/AllowedDomainsPanel/index.tsx:22]: Added validation in the API

File: packages/backend/src/models/OrganizationAllowedEmailDomainsModel.ts [hunk 2]
```diff
@@ -23,7 +27,11 @@ export class OrganizationAllowedEmailDomainsModel {
             organizationUuid:
                 dbOrganizationAllowedEmailDomains.organization_uuid,
             emailDomains: dbOrganizationAllowedEmailDomains.email_domains,
-            role: dbOrganizationAllowedEmailDomains.role,
+            role: isAllowedEmailDomainsRole(
```
Comment (owlas) [inline packages/backend/src/models/OrganizationAllowedEmailDomainsModel.ts:30]: Do we need this check? I assumed the controller should do the validation for us

Comment (ZeRego) [reply packages/backend/src/models/OrganizationAllowedEmailDomainsModel.ts:30]: The column allows any org type. I could have added a migration to create a new table and change this column reference but felt like overkill.

**Review by owlas — APPROVED** [review]

**Pull request merged**